src/transformers/generation_utils.py

```diff
@@ -712,15 +713,33 @@ def _get_logits_processor(
             processors.append(ForcedEOSTokenLogitsProcessor(max_length, forced_eos_token_id))
         if remove_invalid_values is True:
             processors.append(InfNanRemoveLogitsProcessor())
+        processors = self._merge_criteria_processor_list(processors, logits_processor)
         return processors
 
-    def _get_stopping_criteria(self, max_length: Optional[int], max_time: Optional[float]) -> StoppingCriteriaList:
-        stopping_criteria = StoppingCriteriaList()
+    def _get_stopping_criteria(
+        self, max_length: Optional[int], max_time: Optional[float], stopping_criteria: Optional[StoppingCriteriaList]
+    ) -> StoppingCriteriaList:
+        criteria = StoppingCriteriaList()
         if max_length is not None:
-            stopping_criteria.append(MaxLengthCriteria(max_length=max_length))
+            criteria.append(MaxLengthCriteria(max_length=max_length))
         if max_time is not None:
-            stopping_criteria.append(MaxTimeCriteria(max_time=max_time))
-        return stopping_criteria
+            criteria.append(MaxTimeCriteria(max_time=max_time))
+        criteria = self._merge_criteria_processor_list(criteria, stopping_criteria)
+        return criteria
+
+    def _merge_criteria_processor_list(
+        self, default_list: StoppingCriteriaList, custom_list: StoppingCriteriaList
+    ) -> StoppingCriteriaList:
+        if custom_list is None:
```

Comment:
How can the custom_list be None? I think it can only be empty no? Wouldn't it be better to do:

```python
if len(custom_list) == 0:
    return default_list
```

Reply:
The default in `generate` is `None` but I can change it to an empty list if you think that is more consistent.

Reply:
An empty `StoppingCriteriaList` and `LogitsProcessorList` to be more precise.